api/v1alpha1/openapi.yaml

```diff
@@ -2502,6 +2511,254 @@ components:
+    DeviceConfigSummaryStatusType:
+      type: string
+      enum:
+      - 'UpToDate'
+      - 'OutOfDate'
+      - 'Updating'
+      - 'Unknown'
+      x-enum-varnames:
+      - 'DeviceConfigSummaryStatusUpToDate'
+      - 'DeviceConfigSummaryStatusOutOfDate'
+      - 'DeviceConfigSummaryStatusUpdating'
+      - 'DeviceConfigSummaryStatusUnknown' 
+    DeviceIntegrityStatus:
+      type: object
+      required:
+        - summary
+      properties:
+        summary:
+          $ref: '#/components/schemas/DeviceIntegrityStatusSummary'
+          description: 'Summary status of the device system integrity.'
+    DeviceIntegrityStatusSummary:
+      type: object
+      required:
+        - status
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceIntegrityStatusSummaryType'
+          description: 'Summary status of the device system integrity.'
+        info:
+          type: string
+          description: 'Human readable information about the last integrity transition.'
+    DeviceIntegrityStatusSummaryType:
+      type: string
+      enum:
+      - 'Passed'
+      - 'Failed'
+      - 'Unknown'
+      - 'Unsupported'
+      x-enum-varnames:
+      - 'DeviceIntegrityStatusPassed'
+      - 'DeviceIntegrityStatusFailed'
+      - 'DeviceIntegrityStatusUnknown'
+      - 'DeviceIntegrityStatusUnsupported'
+    DeviceResourceStatus:
+      type: object
+      required:
+        - summary
+      properties:
+        cpu:
+          $ref: '#/components/schemas/DeviceResourceStatusType'
+          description: 'Status of the device CPU resources.'
+        memory:
+          $ref: '#/components/schemas/DeviceResourceStatusType'
+          description: 'Status of the device memory resources.'
+        disk:
+          $ref: '#/components/schemas/DeviceResourceStatusType'
+          description: 'Status of the device disk resources.'
+        summary:
+          $ref: '#/components/schemas/DeviceResourceStatusSummary'
+          description: 'Summary status of the device resources.'
+    DeviceResourceStatusSummary:
+      type: object
+      required:
+        - status
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceResourceStatusType'
+          description: 'Status of the device resources.'
+        info:
+          type: string
+          description: 'Human readable information about the last resource transition.'
+    DeviceResourceStatusType:
+      type: string
+      enum:
+      - 'Healthy'
+      - 'Warning'
+      - 'Critical'
+      - 'Unknown'
+      x-enum-varnames:
+      - 'DeviceResourceStatusHealthy'
+      - 'DeviceResourceStatusWarning'
+      - 'DeviceResourceStatusCritical'
+      - 'DeviceResourceStatusUnknown'
+    DeviceConfigStatus:
+      type: object
+      required:
+        - osImage
+        - summary
+      properties:
+        osImage:
+          type: string
+          description: 'The current device Os image.'
+        templateVersion:
```

Comment:
Yeah this ended up being forced and I agree suboptimal. In general an update is considered like a transaction if any part fails all changes are rolled back.

In that case the only value that needs tracked is renderedVersion right. Since the os etc is declared why does the agent need to reflect that data. The service already has it.

I guess if the service would like to write the field that is another question. Does it concern you that it appears to be a growing number of fields being written to device status by the service? Will that have performance implications long term?

Reply:
added renderedVersion to root